### ISSUE TITLE
rare categories give a warning

### ISSUE THREAD

**Issue by mikebenfield** [Microsoft/LightGBM#130]
When I try to mark a feature as categorical, but certain categories are rare, I often get a warning. Some example Python code:

```
import numpy as np
from lightgbm import LGBMClassifier

X = np.ones([50, 1], dtype=np.int)
X[0, 0] = 0
y = np.random.randint(0, 2, size=len(X))
model = LGBMClassifier()
model.fit(X, y, categorical_feature=[0])
```

The warning is:
```
[LightGBM] [Info] Warning: sparse feature with negative values, treating negative values as zero
````

**Comment by gugatr0n1c** [comment]
I get same warning even without marking features as categorical... what does this means?

**Comment by guolinke** [comment]
For categorical feature, it is a wrong warning. I forget to remove it... will fix soon.

@gugatr0n1c For numerical feature, you can pass an additional parameter {"is_sparse":False} to quick fix. And I think this can be fixed with #122

**Comment by gugatr0n1c** [comment]
@guolinke it was working, but in latest version it appears again, even with {"is_sparse":False}

**Comment by guolinke** [comment]
@gugatr0n1c Did you use sklearn interface?
You can try the latest version, i just commit a fix

**Comment by gugatr0n1c** [comment]
Thx, but still same problem. No, I am not using sklearn.
Calling this:

boost_type = 'gbdt'
    if dart_rate > 0:
        boost_type = 'dart'
                
    params = {
                        'task' : 'train',
                        'boosting_type' : boost_type,
                        'objective' : 'regression',
                        'metric' : 'l2',
                        'max_depth' : int(max_depth),
                        'num_leaves' : int(list_count),
                        'min_data_in_leaf' : int(min_in_leaf),
                        'learning_rate' : lr,
                        'feature_fraction' : subF,
                        'bagging_fraction' : subR,
                        'bagging_freq': 1,
                        'verbose' : 0,
                        'nthread' : nthread,
                        'is_sparse' : False,
                        'drop_rate': dart_rate,
                        'max_bin': max_bin,
                        'lambda_l1' : lambda_l1,
                        'lambda_l2' : lambda_l2,
                        'min_gain_to_split' : min_gain_to_split,
                        'min_sum_hessian_in_leaf' : min_sum_hessian_in_leaf
            }
    model = lg.train(
                        params,
                        lg.Dataset(matrix_learn, target_learn),
                        num_boost_round = iterace,
                        valid_sets = lg.Dataset(matrix_test, target_test),
                        early_stopping_rounds = 50
                    )

**Comment by guolinke** [comment]
@gugatr0n1c you should pass "params"  to Dataset as well. 
"is_sparse" is parameter for Dataset

**Comment by gugatr0n1c** [comment]
oh I see, thx, yeap, working now

**Comment by geoHeil** [comment]
How can I set `is_sparse` via sklearn?
